infra/modules/azure_function_app/main.tf

```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.103.1"
```

Comment:
Is there any feature that forces to have the azurerm provider constrained to a very recent version?